metadata-ingestion/src/datahub/utilities/sql_parser.py

```diff
@@ -72,3 +83,116 @@ def get_columns(self) -> List[str]:
+        sql_query = re.sub(
+            r"(\btimestamp\b)",
+            rf"{self._TIMESTAMP_SWAP_TOKEN}",
+            sql_query,
+            flags=re.IGNORECASE,
+        )
+
+        # MetadataSQLParser does not handle "encode" directives well. Remove them
+        sql_query = re.sub(r"\sencode [a-zA-Z]*", "", sql_query, flags=re.IGNORECASE)
+
+        # Replace lookml templates with the variable otherwise sqlparse can't parse ${
+        sql_query = re.sub(r"(\${)(.+)(})", r"\2", sql_query)
+        if sql_query != original_sql_query:
+            logger.debug(f"rewrote original query {original_sql_query} as {sql_query}")
+
+        self._sql = sql_query
+
+        self._stmt = [
+            s
+            for s in sqlparse.parse(
+                # first apply sqlparser formatting just to get rid of comments, which cause
+                # inconsistencies in parsing output
+                sqlparse.format(
+                    self._sql.strip(),
+                    strip_comments=True,
+                    use_space_around_operators=True,
+                ),
+            )
+            if s.token_first(skip_cm=True)
+        ]
+
+        with unittest.mock.patch(
+            "sqllineage.core.handlers.source.SourceHandler.end_of_query_cleanup",
+            datahub.utilities.sqllineage_patch.end_of_query_cleanup_patch,
+        ):
+            with unittest.mock.patch(
+                "sqllineage.core.holders.SubQueryLineageHolder.add_column_lineage",
+                datahub.utilities.sqllineage_patch.add_column_lineage_patch,
+            ):
+                self._stmt_holders = [
+                    LineageAnalyzer().analyze(stmt) for stmt in self._stmt
+                ]
+                self._sql_holder = SQLLineageHolder.of(*self._stmt_holders)
+
+    def get_tables(self) -> List[str]:
+        result: List[str] = list()
+        for table in self._sql_holder.source_tables:
+            table_normalized = re.sub(r"^<default>.", "", str(table))
+            result.append(str(table_normalized))
+
+        # Sort tables to make the list deterministic
+        result.sort()
+
+        return result
+
+    def get_columns(self) -> List[str]:
+        graph: DiGraph = self._sql_holder.graph  # For mypy attribute checking
+        column_nodes = [n for n in graph.nodes if isinstance(n, Column)]
+        column_graph = graph.subgraph(column_nodes)
+
+        target_columns = {column for column, deg in column_graph.out_degree if deg == 0}
+
+        result: Set[str] = set()
+        for column in target_columns:
+            # Let's drop all the count(*) and similard columns which are expression actually if it does not have an alias
+            if not any(ele in column.raw_name for ele in ["*", "(", ")"]):
+                result.add(str(column.raw_name))
+
+        # Reverting back all the previously renamed words which confuses the parser
+        result = set(["date" if c == self._DATE_SWAP_TOKEN else c for c in result])
```

Comment:
There is one iteration over the entire result set for replacing each swapped token. Could you replace lines 176-182 with a single statement like
 ```python 
result = set([unswap_tokens(c) for c in result])
```
where `unswap_tokens` handles the token conversion as needed?